pkg/blob/azure.go

```diff
@@ -39,28 +39,30 @@ import (
 var (
 	DefaultAzureCredentialFileEnv = "AZURE_CREDENTIAL_FILE"
 	DefaultCredFilePath           = "/etc/kubernetes/azure.json"
+	storageEndpointNameToSuffix   = map[string]string{
+		"publiccloud":       "core.windows.net",
```

Comment:
if it's only endpoint name, why not let user provide end point by themselves, by default it's always `core.windows.net`

Reply:
another way is provide a new parameter(`volumeAttributes.storageEndpoint`), that would be more flexible, even node has `azure.json`, if user provides `storageEndpoint` in storage class, should use this value